hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -660,10 +695,11 @@ private void closeOrCancelOpenDeviceActions(final List<JpaAction> actions){
         }
     }
 
-    private List<JpaAction> activateActions(final List<JpaAction> actions){
+    private List<JpaAction> activateActions(final List<JpaAction> actions) {
         actions.forEach(action -> {
             action.setActive(true);
-            final boolean confirmationRequired = action.getRolloutGroup().isConfirmationRequired();
+            final boolean confirmationRequired = action.getRolloutGroup().isConfirmationRequired()
+                    && action.getTarget().getAutoConfirmationStatus() == null;
```

Comment:
At this point we are checking if the confirmation is required and make the check if auto confirmation is active. Only if auto-conf deactivated and confirmation required case we should put the action in WFC state. Otherwise, we will put it directly in the RUNNING state.